src/services/appointment.service.ts

```diff
@@ -0,0 +1,44 @@
+import moment from 'moment';
+import { IAppointment } from '../models/Appointment';
+import { Appointment } from '../models';
+
+const getAppointment = async (userId: string, role: string, appointmentId: string): Promise<IAppointment | null> => {
+  const query = {
+    ...(role === 'patient' ? { patientId: userId } : { doctorId: userId }),
+    _id: appointmentId,
+  };
+
+  const appointment = await Appointment.findOne(query);
+  return appointment;
+};
+
+const getAppointments = async (
+  userId: string,
+  role: string,
+  status = 'active',
+  isToday = false,
+  name?: string,
+): Promise<IAppointment[] | null> => {
+  let query = {};
+  query = {
+    ...(role === 'patient' ? { patientId: userId } : { doctorId: userId }),
+    ...(status && { status }),
+    ...(name && { name }),
+  };
+
+  if (isToday) {
```

Comment:
giả sử t đặt lịch cho ngày mai thì vẫn hiện chứ cần chi đúng ngày Việt hè

Reply:
t để thêm filter để nếu họ muốn xem ngày hôm nay họ có những lịch nào thì sẽ filter được

Reply:
cái này có thể cho lên frontend luôn cũng được, k cần thiết để ở BE đâu nha

Reply:
ời rk m cứ theo ý m